CDCgov/prime-simplereport
# Add frontend support for Manage Users pagination

# FRONTEND PULL REQUEST

## Related Issue

- Resolves #8103 

## Changes Proposed

- Adds pagination to the Manage Users side nav
- Updates the routing for the default Settings page to use `settings/users/:pageNumber` so that the page number can be retrieved from the url params
- Adds a "No results found" message with a button to clear the search filter

## Testing

- Deployed on dev5
- Select `Access organization account` from the support admin dashboard
- Select `Big Organization`, type in "test" for justification, and click `Access data`
- Select either facility
- Go to the Settings page and it should bring you to Manage Users
- Test that pagination works as expected

## Screenshots

Page 1 of results
![image](https://github.com/user-attachments/assets/53e837ab-e60c-4d69-9fa6-e5b02ef0b470)

Page 5 of results
![image](https://github.com/user-attachments/assets/c672270b-e953-4efc-996b-8eedac4ecfa0)

Last page of results
![image](https://github.com/user-attachments/assets/b5ec8ce8-44a7-4ee8-8ea3-8e5859a8cf22)

Search results with multiple pages
![image](https://github.com/user-attachments/assets/4c3606a6-55cb-4e4c-ba38-dbbf4813c54c)

Search results with single result page
![image](https://github.com/user-attachments/assets/dd0da0b1-d76a-46ba-8e5a-9e2dd3492143)

No results found
![image](https://github.com/user-attachments/assets/012ccb86-f2e8-4a2f-b777-042cf2089e05)
